smse/pipelines/image.py

```diff
@@ -42,25 +42,80 @@ def load(self, input_path: Union[str, Path]) -> ImageT:
     def validate(self, data: Any) -> bool:
         return isinstance(data, np.ndarray) and len(data.shape) in [2, 3]
 
-    def preprocess(self, image: ImageT) -> ImageT:
-        """Preprocess image data"""
+    def default_mode(self, image: ImageT) -> ImageT:
+        """Converting image to default mode BGR"""
         import cv2  # type: ignore[import-not-found]
 
-        # Resize
-        image = cv2.resize(image, self.config.target_size)
+        if not isinstance(image, np.ndarray):  # Assuming PIL Image
+            image = np.array(image)
+
+        if len(image.shape) == 2:  # Grayscale
+            converted_image = cv2.cvtColor(image, cv2.COLOR_GRAY2BGR)
+        elif image.shape[2] == 3:  # RGB or BGR, ensure BGR.
+            converted_image = cv2.cvtColor(image, cv2.COLOR_RGB2BGR)
+        elif image.shape[2] == 4:  # RGBA
+            converted_image = cv2.cvtColor(image, cv2.COLOR_RGBA2BGR)
+        else:
+            raise ValueError(f"Unsupported image shape: {image.shape}")
+
+        return cast(ImageT, converted_image)
+
+    def preprocess(
+        self,
+        image: ImageT,
+        resizer: Union[bool, tuple[int, int]] = False,
+        mode_converter: int = 0,
+        median_blur: bool = False,
+        normalizer: bool = False,
+    ) -> ImageT:
```

Comment:
This function should also accept a batch of images or make a new function like `preprocess_batch` that utilizes `preprocess` to work on a batch of images